lib/recurly/hosted-fields.js

```diff
@@ -96,17 +101,94 @@ export class HostedFields extends Emitter {
     this.initQueue = [];
   }
 
-  readyHandler (body) {
+  onReady (body) {
     const pos = this.initQueue.indexOf(body.type);
     if (~pos) this.initQueue.splice(pos, 1);
     if (this.initQueue.length === 0) {
-      this.off('hostedField:ready', this.readyHandler);
+      this.off('hostedField:ready', this.onReady);
       this.bus.send('hostedFields:ready');
       this.ready = true;
     }
     this.update(body);
   }
 
+  onTab (direction, message) {
+    const origin = this.getFieldByType(message.type);
+    let tabbableItems = tabbable(global.document.body);
+
+    // Add hosted fields to the tabbable list
+    this.fields.forEach(field => {
+      if (field.container.compareDocumentPosition(tabbableItems[0]) & Node.DOCUMENT_POSITION_FOLLOWING) {
+        tabbableItems.unshift(field);
+      } else if (field.container.compareDocumentPosition(last(tabbableItems)) & Node.DOCUMENT_POSITION_PRECEDING) {
+        tabbableItems.push(field);
+      } else {
+        tabbableItems.find((item, i) => {
+          let el = item instanceof HostedField ? item.container : item;
+          let nextItem = tabbableItems[i + 1];
+          let nextEl = nextItem instanceof HostedField ? nextItem.container : nextItem;
+          if (!dom.element(el)) return;
+          if (!dom.element(nextEl)) return;
+          if (!(field.container.compareDocumentPosition(el) & Node.DOCUMENT_POSITION_PRECEDING)) return;
+          if (!(field.container.compareDocumentPosition(nextEl) & Node.DOCUMENT_POSITION_FOLLOWING)) return;
+          tabbableItems.splice(i + 1, 0, field);
+          return true;
+        });
+      }
+    });
```

Comment:
I don't think this is necessary since there are input proxies being injected before the hosted field iframes.

Reply:
@chrissrogers we talked about this previously but I'd like to re-emphasize the issue again.

I attempted to undo the above changes and rely strictly on the hidden inputs to determine the adjacent iframe input for focus. However, in trying to reference the input within the iframe, I was receiving the infamous `DOM Exception 18` which does not permit access to elements within an iframe if the iframe src and its parent src differ.

Reply:
Per discussion, the focus intent should be implemented by message bus and not through direct DOM access. 👍